cli/command/swarm/unlock.go

```diff
@@ -24,6 +26,14 @@ func newUnlockCommand(dockerCli *command.DockerCli) *cobra.Command {
 			client := dockerCli.Client()
 			ctx := context.Background()
 
+			// first see if the swarm is actually locked by running a node LS first, and bail
+			// if we don't get an error.  If we get any error, such as the swarm manager being
+			// locked, proceed anyway
+			_, err := client.NodeList(ctx, types.NodeListOptions{})
+			if err == nil {
+				return errors.New("Error: swarm is not locked")
+			}
```

Comment:
This would probably solve my comment https://github.com/docker/docker/pull/29470#issuecomment-267565880 as well, because we'd be able to check _both_ if it's a swarm manager, _and_ if it's locked